pyomo/contrib/solver/base.py

```diff
@@ -393,8 +406,29 @@ def _map_config(
             self.config.time_limit = timelimit
         if report_timing is not NOTSET:
             self.config.report_timing = report_timing
-        if options is not NOTSET:
+        if hasattr(self, 'options'):
+            self.config.solver_options.set_value(self.options)
+        if hasattr(self, 'solver_options'):
+            self.config.solver_options.set_value(self.solver_options)
+        if (options is not NOTSET) and (solver_options is not NOTSET):
+            # There is no reason for a user to be trying to mix both old
+            # and new options. That is silly. So we will yell at them.
+            # Example that would raise an error:
+            # solver.solve(model, options={'foo' : 'bar'}, solver_options={'foo' : 'not_bar'})
+            raise ApplicationError(
+                "Both 'options' and 'solver_options' were declared "
+                "in the 'solve' call. Please use one or the other, "
```

Comment:
Isn't `_map_config` also called from set_options? Is referencing "the solve call" here too specific?